lib/Twig/Node/Module.php

```diff
@@ -94,7 +94,9 @@ protected function compileTemplate(Twig_Compiler $compiler)
 
         $this->compileDebugInfo($compiler);
 
-        $this->compileGetSource($compiler);
+        if ($compiler->getEnvironment()->isDebug()) {
+            $this->compileGetSource($compiler);
+        }
```

Comment:
I would always compile the `getSource()` method but make it return an empty string if debug is disabled.